### PR TITLE
Add script(s) for issue 3674

### PR DESCRIPTION
**Pull request by VNikiforov-dev**
ATF Test Scripts to check #[3674]

This PR is **not ready for Internal** review.

### Summary
Scripts to check 3674 issue

### ATF version
develop

### Changelog
- Added scripts to check 3674 issue

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/8_0/3674_SDL_sends_OnAppRegistered_before_app_has_valid_internal_HMI_state.lua [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    local appParams = utils.cloneTable(common.app.getParams())
+    common.getMobileSession():SendRPC("RegisterAppInterface", appParams)
+
+    common.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered")
+    :Do(function ()
+      common.getHMIConnection():SendNotification("BasicCommunication.OnSystemRequest", sysRequestParams)
+    end)
+
+    end)
+end
```
Comment (dboltovskyi) [inline test_scripts/Defects/8_0/3674_SDL_sends_OnAppRegistered_before_app_has_valid_internal_HMI_state.lua:49]: Please correct indentations in this function

File: test_scripts/Defects/8_0/3674_SDL_sends_OnAppRegistered_before_app_has_valid_internal_HMI_state.lua [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  :Do(function()
+
+  common.getMobileSession():ExpectNotification("OnSystemRequest"):Times(2)
+  :ValidIf(function (_, data)
+    if data.payload.requestType == "PROPRIETARY" then return true end
+    if data.payload.requestType == "LOCK_SCREEN_ICON_URL" then return true end
+    return false
+    end)
+
+    local appParams = utils.cloneTable(common.app.getParams())
```
Comment (dboltovskyi) [inline test_scripts/Defects/8_0/3674_SDL_sends_OnAppRegistered_before_app_has_valid_internal_HMI_state.lua:40]: Using of `cloneTalbe()` is not required

File: test_scripts/Defects/8_0/3674_SDL_sends_OnAppRegistered_before_app_has_valid_internal_HMI_state.lua [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+--[[ Local function ]]
+local function OnSystemRequestDuringRegistationApp()
+  common.getMobileSession():StartService(7)
+  :Do(function()
+
+  common.getMobileSession():ExpectNotification("OnSystemRequest"):Times(2)
+  :ValidIf(function (_, data)
+    if data.payload.requestType == "PROPRIETARY" then return true end
+    if data.payload.requestType == "LOCK_SCREEN_ICON_URL" then return true end
+    return false
```
Comment (dboltovskyi) [inline test_scripts/Defects/8_0/3674_SDL_sends_OnAppRegistered_before_app_has_valid_internal_HMI_state.lua:37]: Please provide the reason of false

File: test_scripts/Defects/8_0/3674_SDL_sends_OnAppRegistered_before_app_has_valid_internal_HMI_state.lua [hunk 4]
```diff
@@ -0,0 +1,60 @@
+---------------------------------------------------------------------------------------------------
+-- Issue: https://github.com/smartdevicelink/sdl_core/issues/3674
+---------------------------------------------------------------------------------------------------
+-- Description: Core sends OnAppRegistered before app has valid internal HMI state
+---------------------------------------------------------------------------------------------------
+-- In case:
+-- 1. App sends RegisterAppInterface
+-- 2. HMI receive BC.OnAppRegistered
```
Comment (dboltovskyi) [inline test_scripts/Defects/8_0/3674_SDL_sends_OnAppRegistered_before_app_has_valid_internal_HMI_state.lua:8]: `receive` => `receives`

**Review by dboltovskyi — CHANGES_REQUESTED** [review]

File: test_scripts/Defects/8_0/3674_SDL_sends_OnAppRegistered_before_app_has_valid_internal_HMI_state.lua [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  requestType = "PROPRIETARY",
+  fileType = "JSON",
+  fileName = "sdl_snapshot.json"
+}
+
+--[[ Local function ]]
+local function notificationDuringRegistationApp()
+  common.getMobileSession():StartService(7)
+  :Do(function()
+    common.getMobileSession():ExpectNotification("OnSystemRequest"):Times(2)
```
Comment (VjKlepikov) [inline test_scripts/Defects/8_0/3674_SDL_sends_OnAppRegistered_before_app_has_valid_internal_HMI_state.lua:31]: Please align
```suggestion
     common.getMobileSession():ExpectNotification("OnSystemRequest"):Times(2)
```

**Review by VjKlepikov — CHANGES_REQUESTED** [review]

**Review by dboltovskyi — APPROVED** [review]

**Comment by dboltovskyi** [comment]
Moved to `https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2593`

**Pull request closed without merging**